src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,429 @@
+        : aacEncOpen(nullptr),
+          aacEncClose(nullptr),
+          aacEncEncode(nullptr),
+          aacEncInfo(nullptr),
+          aacEncoder_SetParam(nullptr),
+          m_aacAot(AOT_AAC_LC),
+          m_bitrate(0),
+          m_channels(0),
+          m_samplerate(0),
+          m_pCallback(pCallback),
+          m_library(nullptr),
+          m_pInputFifo(nullptr),
+          m_pFifoChunkBuffer(nullptr),
+          m_readRequired(0),
+          m_aacEnc(),
+          m_pAacDataBuffer(nullptr),
+          m_aacInfo(),
+          m_hasSbr(false) {
+    // Load shared library
+    // Code import from encodermp3.cpp
+    QStringList libnames;
+#ifdef __LINUX__
+    libnames << "fdk-aac";
+    libnames << "libfdk-aac.so.2";
+    libnames << "libfdk-aac.so.1";
+#elif __WINDOWS__
+    // Give top priority to libfdk-aac copied
+    // into Mixxx's installation folder
+    libnames << "libfdk-aac-2.dll";
+    libnames << "libfdk-aac-1.dll";
+
+    // Fallback and user-friendly method: use libfdk-aac
+    // provided with B.U.T.T installed in
+    // a standard location
+    QString buttFdkAacPath = buttWindowsFdkAac();
+    if (!buttFdkAacPath.isNull()) {
+        kLogger.debug() << "Found libfdk-aac at" << buttFdkAacPath;
+        libnames << buttFdkAacPath;
+    }
+
+    // Last resort choices: try versions with unusual names
+    libnames << "libfdk-aac.dll";
+    libnames << "libfdkaac.dll";
+#elif __APPLE__
+    // Using Homebrew ('brew install fdk-aac' command):
+    libnames << "/usr/local/lib/libfdk-aac.dylib";
+    libnames << "/usr/local/lib/libfdk-aac.2.dylib";
+    libnames << "/usr/local/lib/libfdk-aac.1.dylib";
+    // Using MacPorts ('sudo port install libfdk-aac' command):
+    libnames << "/opt/local/lib/libfdk-aac.dylib";
+    libnames << "/opt/local/lib/libfdk-aac.2.dylib";
+    libnames << "/opt/local/lib/libfdk-aac.1.dylib";
+#endif
+
+    QString failedMsg = "Failed to load AAC encoder library";
+    for (const auto& libname : qAsConst(libnames)) {
+        m_library = new QLibrary(libname);
+        if (m_library->load()) {
+            kLogger.debug() << "Successfully loaded encoder library " << m_library->fileName();
+            break;
+        } else {
+            // collect error messages for the case we have no success
+            failedMsg.append("\n" + m_library->errorString());
+        }
+        delete m_library;
+        m_library = nullptr;
+    }
+
+    if (!m_library || !m_library->isLoaded()) {
+        kLogger.warning() << failedMsg;
+        return;
+    }
+
+    aacEncGetLibInfo = (aacEncGetLibInfo_)m_library->resolve("aacEncGetLibInfo");
+    aacEncOpen = (aacEncOpen_)m_library->resolve("aacEncOpen");
+    aacEncClose = (aacEncClose_)m_library->resolve("aacEncClose");
+    aacEncEncode = (aacEncEncode_)m_library->resolve("aacEncEncode");
+    aacEncInfo = (aacEncInfo_)m_library->resolve("aacEncInfo");
+    aacEncoder_SetParam = (aacEncoder_SetParam_)m_library->resolve("aacEncoder_SetParam");
```

Comment:
QStringLiteral

Reply:
Not in this case: 
QLibrary::resolve(const char *symbol)